[BUG]: test_torch_composite_loss and test_torch_focal_loss fail

_Thank you for helping us improve `solaris`!_

## Summary of the bug

Two of the Travis tests now fail that worked previously (test_torch_composite_loss and test_torch_focal_loss fail).

## Steps to reproduce the bug
Run Travis on dev branch.

## Buggy behavior and/or error message

=================================== FAILURES ===================================
____________________ TestGetLoss.test_torch_composite_loss _____________________
self = <test_losses.TestGetLoss object at 0x7fd6576c1790>
    def test_torch_composite_loss(self):
        epsilon = 1e-4
        loss_dict = {'bce' : {}, 'hinge' : {}}
        lf = get_loss('torch', loss_dict)
        y_true = torch.tensor([1, 1, 1], dtype=torch.float)
        y_pred = torch.tensor([0, 1, 0], dtype=torch.float)
>       assert np.abs(
            lf.forward(y_true, y_pred) - 19.4207) < epsilon
E       AssertionError: assert tensor(48.2460) < 0.0001
E        +  where tensor(48.2460) = <ufunc 'absolute'>((tensor(67.6667) - 19.4207))
E        +    where <ufunc 'absolute'> = np.abs
E        +    and   tensor(67.6667) = <bound method TorchCompositeLoss.forward of TorchCompositeLoss()>(tensor([1., 1., 1.]), tensor([0., 1., 0.]))
E        +      where <bound method TorchCompositeLoss.forward of TorchCompositeLoss()> = TorchCompositeLoss().forward
tests/test_nets/test_losses.py:41: AssertionError
_________________ TestTorchCustomLosses.test_torch_focal_loss __________________
self = <test_losses.TestTorchCustomLosses object at 0x7fd65756c050>
    def test_torch_focal_loss(self):
        epsilon = 1e-6
        y_true = torch.tensor([1, 1, 1], dtype=torch.float)
        y_pred = torch.tensor([0, 1, 0], dtype=torch.float)
        lf = TorchFocalLoss()
>       assert np.abs(
            lf.forward(y_pred, y_true) - 18.420681) < epsilon
E       AssertionError: assert tensor(48.2460) < 1e-06
E        +  where tensor(48.2460) = <ufunc 'absolute'>((tensor(66.6667) - 18.420681))
E        +    where <ufunc 'absolute'> = np.abs
E        +    and   tensor(66.6667) = <bound method TorchFocalLoss.forward of TorchFocalLoss()>(tensor([0., 1., 0.]), tensor([1., 1., 1.]))
E        +      where <bound method TorchFocalLoss.forward of TorchFocalLoss()> = TorchFocalLoss().forward
tests/test_nets/test_losses.py:82: AssertionError

## Expected behavior

Tests should run without error.

## Environment information
- `solaris` version: dev branch
- python version: 3.6, 3.7, 3.8
- version of any relevant dependencies (optional - we may ask for this information later if not provided)

--- a/solaris/nets/_torch_losses.py
+++ b/solaris/nets/_torch_losses.py
@@ -65,10 +65,10 @@ class TorchFocalLoss(nn.Module):
 
         if self.logits:
             BCE_loss = F.binary_cross_entropy_with_logits(outputs, targets,
-                                                          reduction='sum')
+                                                          reduction='none')
         else:
             BCE_loss = F.binary_cross_entropy(outputs, targets,
-                                              reduction='sum')
+                                              reduction='none')
         pt = torch.exp(-BCE_loss)
         F_loss = (1-pt)**self.gamma * BCE_loss
         if self.reduce:

--- a/solaris/nets/_torch_losses.py
+++ b/solaris/nets/_torch_losses.py
@@ -64,9 +64,11 @@ class TorchFocalLoss(nn.Module):
         """
 
         if self.logits:
-            BCE_loss = F.binary_cross_entropy_with_logits(outputs, targets)
+            BCE_loss = F.binary_cross_entropy_with_logits(outputs, targets,
+                                                          reduction='none')
         else:
-            BCE_loss = F.binary_cross_entropy(outputs, targets)
+            BCE_loss = F.binary_cross_entropy(outputs, targets,
+                                              reduction='none')
         pt = torch.exp(-BCE_loss)
         F_loss = (1-pt)**self.gamma * BCE_loss
         if self.reduce:

--- a/solaris/nets/_torch_losses.py
+++ b/solaris/nets/_torch_losses.py
@@ -65,10 +65,10 @@ class TorchFocalLoss(nn.Module):
 
         if self.logits:
             BCE_loss = F.binary_cross_entropy_with_logits(outputs, targets,
-                                                          reduction='mean')
+                                                          reduction='sum')
         else:
             BCE_loss = F.binary_cross_entropy(outputs, targets,
-                                              reduction='mean')
+                                              reduction='sum')
         pt = torch.exp(-BCE_loss)
         F_loss = (1-pt)**self.gamma * BCE_loss
         if self.reduce:

--- a/solaris/nets/_torch_losses.py
+++ b/solaris/nets/_torch_losses.py
@@ -65,10 +65,10 @@ class TorchFocalLoss(nn.Module):
 
         if self.logits:
             BCE_loss = F.binary_cross_entropy_with_logits(outputs, targets,
-                                                          reduction='none')
+                                                          reduction='mean')
         else:
             BCE_loss = F.binary_cross_entropy(outputs, targets,
-                                              reduction='none')
+                                              reduction='mean')
         pt = torch.exp(-BCE_loss)
         F_loss = (1-pt)**self.gamma * BCE_loss
         if self.reduce:

--- a/tests/test_nets/test_losses.py
+++ b/tests/test_nets/test_losses.py
@@ -20,12 +20,12 @@ class TestGetLoss(object):
         epsilon = 1e-6
         loss_dict = {'bce' : {}, 'hinge' : {}}
         lf = get_loss('keras', loss_dict)
-        y_true = tf.constant([1, 1, 1], dtype='float')
-        y_pred = tf.constant([0, 1, 0], dtype='float')
+        y_true = tf.constant([0, 1, 1], dtype='float')
+        y_pred = tf.constant([.1, .9, .4], dtype='float')
         sess = tf.Session()
         with sess.as_default():
             assert np.abs(
-                lf(y_true, y_pred).eval() - 11.41206380063888) < epsilon
+                lf(y_true, y_pred).eval() - 8.3682727814) < epsilon
 
     def test_torch_vanilla_loss(self):
         loss_dict = {'bce' : {}}
@@ -36,36 +36,36 @@ class TestGetLoss(object):
         epsilon = 1e-4
         loss_dict = {'bce' : {}, 'hinge' : {}}
         lf = get_loss('torch', loss_dict)
-        y_true = torch.tensor([1, 1, 1], dtype=torch.float)
-        y_pred = torch.tensor([0, 1, 0], dtype=torch.float)
+        y_true = torch.tensor([0, 1, 1], dtype=torch.float)
+        y_pred = torch.tensor([.1, .9, .4], dtype=torch.float)
         assert np.abs(
-            lf.forward(y_true, y_pred) - 19.4207) < epsilon
+            lf.forward(y_true, y_pred) - 8.3682727814) < epsilon
 
 
 class TestKerasCustomLosses(object):
 
     def test_keras_focal_loss(self):
         epsilon = 1e-6
-        y_true = tf.constant([1, 1, 1], dtype='float')
-        y_pred = tf.constant([0, 1, 0], dtype='float')
+        y_true = tf.constant([0, 1, 1], dtype='float')
+        y_pred = tf.constant([.1, .9, .4], dtype='float')
         sess = tf.Session()
         with sess.as_default():
             foc_loss = k_focal_loss()(y_true, y_pred).eval()
-            assert np.abs(foc_loss - 41.446533) < epsilon
+            assert np.abs(foc_loss - 0.1106572822) < epsilon
 
     def test_keras_lovasz_hinge(self):
         epsilon = 1e-6
-        y_true = tf.constant([1, 1, 1], dtype='float')
-        y_pred = tf.constant([0, 1, 0], dtype='float')
+        y_true = tf.constant([0, 1, 1], dtype='float')
+        y_pred = tf.constant([.1, .9, .4], dtype='float')
         sess = tf.Session()
         with sess.as_default():
             lov_loss = k_lovasz_hinge()(y_true, y_pred).eval()
-            assert np.abs(lov_loss - 19.087347) < epsilon
+            assert np.abs(lov_loss - 0.8517242074) < epsilon
 
     def test_keras_jaccard_loss(self):
         epsilon = 1e-6
-        y_true = tf.constant([1, 1, 1], dtype='float')
-        y_pred = tf.constant([0, 1, 0], dtype='float')
+        y_true = tf.constant([0, 1, 1], dtype='float')
+        y_pred = tf.constant([.1, .9, .4], dtype='float')
         sess = tf.Session()
         with sess.as_default():
             jac_loss = k_jaccard_loss(y_true, y_pred).eval()
@@ -76,24 +76,24 @@ class TestTorchCustomLosses(object):
 
     def test_torch_focal_loss(self):
         epsilon = 1e-6
-        y_true = torch.tensor([1, 1, 1], dtype=torch.float)
-        y_pred = torch.tensor([0, 1, 0], dtype=torch.float)
+        y_true = torch.tensor([0, 1, 1], dtype=torch.float)
+        y_pred = torch.tensor([.1, .9, .4], dtype=torch.float)
         lf = TorchFocalLoss()
         assert np.abs(
-            lf.forward(y_pred, y_true) - 18.420681) < epsilon
+            lf.forward(y_pred, y_true) - 0.1106572822) < epsilon
 
     def test_torch_focal_loss_same(self):
         epsilon = 1e-5
-        y_true = torch.tensor([0, 1, 0], dtype=torch.float)
-        y_pred = torch.tensor([0, 1, 0], dtype=torch.float)
+        y_true = torch.tensor([0, 1, 1], dtype=torch.float)
+        y_pred = torch.tensor([0, 1, 1], dtype=torch.float)
         lf = TorchFocalLoss()
         assert np.abs(
             lf.forward(y_pred, y_true) - 0.) < epsilon
 
     def test_torch_lovasz_hinge(self):
         epsilon = 1e-6
-        y_true = torch.tensor([1, 1, 1], dtype=torch.float)
-        y_pred = torch.tensor([0, 1, 0], dtype=torch.float)
+        y_true = torch.tensor([0, 1, 1], dtype=torch.float)
+        y_pred = torch.tensor([.1, .9, .4], dtype=torch.float)
         lf = torch_lovasz_hinge
         assert np.abs(
-            lf(y_pred, y_true) - 0.6666666269302368) < epsilon
+            lf(y_pred, y_true) - 0.8517242074) < epsilon

--- a/tests/test_nets/test_losses.py
+++ b/tests/test_nets/test_losses.py
@@ -39,7 +39,7 @@ class TestGetLoss(object):
         y_true = torch.tensor([0, 1, 1], dtype=torch.float)
         y_pred = torch.tensor([.1, .9, .4], dtype=torch.float)
         assert np.abs(
-            lf.forward(y_true, y_pred) - 8.3682727814) < epsilon
+            lf.forward(y_pred, y_true) - 1.1423372030) < epsilon
 
 
 class TestKerasCustomLosses(object):
@@ -96,4 +96,4 @@ class TestTorchCustomLosses(object):
         y_pred = torch.tensor([.1, .9, .4], dtype=torch.float)
         lf = torch_lovasz_hinge
         assert np.abs(
-            lf(y_pred, y_true) - 0.8517242074) < epsilon
+            lf(y_pred, y_true) - 0.6000000000) < epsilon

--- a/tests/test_nets/test_losses.py
+++ b/tests/test_nets/test_losses.py
@@ -25,7 +25,7 @@ class TestGetLoss(object):
         sess = tf.Session()
         with sess.as_default():
             assert np.abs(
-                lf(y_true, y_pred).eval() - 8.3682727814) < epsilon
+                lf(y_true, y_pred).eval() - 0.9423373) < epsilon
 
     def test_torch_vanilla_loss(self):
         loss_dict = {'bce' : {}}
@@ -51,7 +51,7 @@ class TestKerasCustomLosses(object):
         sess = tf.Session()
         with sess.as_default():
             foc_loss = k_focal_loss()(y_true, y_pred).eval()
-            assert np.abs(foc_loss - 0.1106572822) < epsilon
+            assert np.abs(foc_loss - 0.24845211) < epsilon
 
     def test_keras_lovasz_hinge(self):
         epsilon = 1e-6
@@ -60,7 +60,7 @@ class TestKerasCustomLosses(object):
         sess = tf.Session()
         with sess.as_default():
             lov_loss = k_lovasz_hinge()(y_true, y_pred).eval()
-            assert np.abs(lov_loss - 0.8517242074) < epsilon
+            assert np.abs(lov_loss - 0.70273256) < epsilon
 
     def test_keras_jaccard_loss(self):
         epsilon = 1e-6
@@ -69,7 +69,7 @@ class TestKerasCustomLosses(object):
         sess = tf.Session()
         with sess.as_default():
             jac_loss = k_jaccard_loss(y_true, y_pred).eval()
-            assert np.abs(jac_loss - 0.66666666667) < epsilon
+            assert np.abs(jac_loss - 0.38095242) < epsilon
 
 
 class TestTorchCustomLosses(object):

